tests/bsim/bluetooth/mesh/src/test_provision.c

```diff
@@ -15,10 +15,6 @@
 
 #if defined CONFIG_BT_MESH_USES_MBEDTLS_PSA
 #include <psa/crypto.h>
-#elif defined CONFIG_BT_MESH_USES_TINYCRYPT
```

Comment:
I guess until Tinycrypt deprecation period is over, the tests should present. @PavelVPV?

Reply:
Since the mesh option stays, yes, we must keep the tests.

Reply:
AFAIK the default policy is not to have *any* in-tree references or tests for deprecated APIs. We can discuss and potentially approve exceptions if there are good justifications for it, but until then we should stick to the normal procedure.

Reply:
Then all bsim tests those previously run with Tinycrypt now run with mbedtls. Furthermore, tests with psa overlay also are built and run with mbedtls. Almost all bsim mesh tests are built and run twice at the moment. That is time and CI loading overkill.

Reply:
> AFAIK the default policy is not to have any in-tree references or tests for deprecated APIs.

Yeah, I this what I understood as well by reading the [API deprecation guide](https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html#deprecated).

> Then all bsim tests those previously run with Tinycrypt now run with mbedtls

Which should be fine, right? I mean, that's the goal of this PR after all.

> Furthermore, tests with psa overlay also are built and run with mbedtls. Almost all bsim mesh tests are built and run twice at the moment.

Right. I saw this and perhaps you already told me this, but I completely forgot to fix it. I will do that ASAP. Thanks!

Reply:
> > AFAIK the default policy is not to have any in-tree references or tests for deprecated APIs.
> 
> Yeah, I this what I understood as well by reading the [API deprecation guide](https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html#deprecated).
> 
> > Then all bsim tests those previously run with Tinycrypt now run with mbedtls
> 
> Which should be fine, right? I mean, that's the goal of this PR after all.
> 
> > Furthermore, tests with psa overlay also are built and run with mbedtls. Almost all bsim mesh tests are built and run twice at the moment.
> 
> Right. I saw this and perhaps you already told me this, but I completely forgot to fix it. I will do that ASAP. Thanks!

Hi @valeriosetti, I do not mean it should be done in this PR. This PR is enormously huge without these changes. However, it should be done.

Reply:
I think part of the confusion here is that there are different interfaces to consider for deprecation and/or removal. The `BT_MESH_USES_TINYCRYPT` Kconfig option can be seen both as an API (something an app makes references to) as well as a user of TinyCrypt. Since TinyCrypt is being deprecated, as a user the option should get removed, however as an API the situation is more murky. I'm not sure there exists clear formal guidance on situations like this, but I'd be in favor of just removing it.

Reply:
We would like to keep both, the `BT_MESH_USES_TINYCRYPT` option and the bsim tests with TinyCrypt until TinyCrypt is completely removed. The reason to keep the tests is to ensure that mesh with TinyCrypt still works and if not, we will fix bugs.

@jhedberg, should I bring this to some meeting?